tests/robustness/client/watch.go

```diff
@@ -20,29 +20,34 @@ import (
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
 
-	"go.etcd.io/etcd/tests/v3/framework/e2e"
 	"go.etcd.io/etcd/tests/v3/robustness/identity"
 	"go.etcd.io/etcd/tests/v3/robustness/report"
 )
 
-func CollectClusterWatchEvents(ctx context.Context, t *testing.T, clus *e2e.EtcdProcessCluster, maxRevisionChan <-chan int64, cfg WatchConfig, baseTime time.Time, ids identity.Provider) []report.ClientReport {
+func CollectClusterWatchEvents(ctx context.Context, lg *zap.Logger, etcdEndpoints [][]string, maxRevisionChan <-chan int64, cfg WatchConfig, baseTime time.Time, ids identity.Provider) ([]report.ClientReport, bool, error) {
 	mux := sync.Mutex{}
 	var wg sync.WaitGroup
-	reports := make([]report.ClientReport, len(clus.Procs))
-	memberMaxRevisionChans := make([]chan int64, len(clus.Procs))
-	for i, member := range clus.Procs {
-		c, err := NewRecordingClient(member.EndpointsGRPC(), ids, baseTime)
-		require.NoError(t, err)
+	neverFailed := true
+	reports := make([]report.ClientReport, len(etcdEndpoints))
+	memberMaxRevisionChans := make([]chan int64, len(etcdEndpoints))
+	for i, endpoint := range etcdEndpoints {
+		c, err := NewRecordingClient(endpoint, ids, baseTime)
```

Comment:
Like mentioned before we don't need to connect to all endpoints from a member. One endpoint per client is enough.